### DISCUSSION TITLE
Commands.json update

### DISCUSSION THREAD

**Discussion by lblum-steno** [opensteno/plover#1676] [Technical Discussion]
Hi there,

When using the commands.json dictionnary I see mapping between, obviosly, steno chords and actions (in windows)
-> Where can I find the naming convention to acceess windows commands? For instance if I wish to add 'page down, page up' or 'shift-tab' commands?

Thanks a lot for your reply/help,

Lionel

**Comment by panathea** [comment]
Check out the Plover wiki page on Keyboard Shortcuts:
https://plover.wiki/index.php/Dictionary_Format#Shortcut_Key_Names

Let us know if you need any help with specific examples of what you're
looking for!

**Reply by lblum-steno** [reply]
Hi Ted,

Thanks a lot for your quick answer.
I checked the page but did not find an exhaustive list of the **control keys** (here is what I am looking for!)

Can you please 2x check?

Many thanks,

Lionel

**Reply by panathea** [reply]
Yeah, I don't see an exhaustive list, and now that the wiki page is a novel-length long anyway, I think we could afford to move all of the missing keys into the table.

Page_Up and Page_Down are the names for those keys, [searching for them](https://github.com/search?q=repo%3Aopenstenoproject%2Fplover+Page_Up&type=code), I find a few places where you can see glimpses of the names.

The [keys layout in plover.machine.keyboard](https://github.com/openstenoproject/plover/blob/9b5a357f1fb57cb0a9a8596ae12cd1e84fcff6c4/plover/machine/keyboard.py#L26) can help with understanding the core key names, but not the number pad:

```
    Escape  F1 F2 F3 F4  F5 F6 F7 F8  F9 F10 F11 F12

      `  1  2  3  4  5  6  7  8  9  0  -  =  \\ BackSpace  Insert Home Page_Up
     Tab  q  w  e  r  t  y  u  i  o  p  [  ]               Delete End  Page_Down
           a  s  d  f  g  h  j  k  l  ;  '      Return
            z  x  c  v  b  n  m  ,  .  /                          Up
                     space                                   Left Down Right
```

Modifiers are: shift, control, alt/option, super/windows/command

**Reply by panathea** [reply]
Just to explicitly answer your initial question: `{#page_up}`, `{#page_down}`, and `{#shift(tab)}`

**Reply by lblum-steno** [reply]
Thank you, I guess I could have deduct it but wanted to avoid guessing (since we are talking about a rather old technologie -> keyboard ;)).

I used the following chords which seems intuitive to me since they are avery similar to my Thinkpad Lenovo keyboard.

STPH-F for page up
STPH-L for page-down

Many thanks!

Lionel
